File: orders-service/models/product.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const ProductSchema = new Schema({
    code: {type:String},
    name: {type:String},
    color: {type:String},
    price: {type:Number},
    quantity: {type:Number}
});

module.exports = mongoose.model('Product', ProductSchema, 'product');

File: catalog-service/models/product.py

# flask_graphene_mongo/models.py
from mongoengine import Document
from mongoengine.fields import (
    StringField, IntField
)


class Product(Document):
    meta = {'collection': 'product'}
    code = StringField(unique=True)
    name = StringField()
    color = StringField()
    price = IntField()
    quantity = IntField()
